wave_lang/kernel/wave/codegen/emitter.py

```diff
@@ -54,8 +54,8 @@
 
 
 def _get_upper_bound(expr: Any) -> Optional[Attribute]:
-    res = subs_idxc(sympy.sympify(expr))
-    if res.is_number:
+    res = subs_idxc(expr)
+    if isinstance(res, int) or res.is_number:
```

Comment:
Why would there be a change here?

Reply:
Fun fact `1` and `sympy.core.numbers.One` will occupy same slot in the dict, so we can end up here with actual integers or symbol literals from the cache. Regardless, refactored it to the standalone func.